### DISCUSSION TITLE
thoughts on supporting astro-web-adapter?

### DISCUSSION THREAD

**Discussion by firxworx** [lukeshay/astro-aws#89] [Ideas]
@lukeshay thank-you for your work on this project it has been very insightful and I found it helpful for my project.

In researching solution paths for hosting node apps + Astro SSR with lambda I came across the aws-lambda-web-adapter and was curious re your thoughts if this would be a good option/direction to support for this project as well? 

https://github.com/awslabs/aws-lambda-web-adapter
https://aws.amazon.com/blogs/compute/using-response-streaming-with-aws-lambda-web-adapter-to-optimize-performance/

It is a general purpose lambda to http api adapter written in Rust. It can be added to Docker images or used with AWS managed runtimes with a few extra steps:

https://github.com/awslabs/aws-lambda-web-adapter#lambda-functions-packaged-as-zip-package-for-aws-managed-runtimes
(also see the example nodejs app: https://github.com/awslabs/aws-lambda-web-adapter/tree/main/examples/expressjs-zip)

The blog article also links to the example project https://github.com/aws-samples/lwa-nextjs-response-streaming-example 
which demonstrates Next13 streaming with suspense w/ lambda response streaming; i.e. use-case relevant to modern SSR.

I guess the cool part of the idea is that Astro could stay pretty idiomatic regarding SSR with only the official nodejs adapter meanwhile the concern of mapping "AWS to http" is handled by an optimized and specialized layer in performant rust.

It would be awesome if the cdk constructs also supported this approach!

**Comment by lukeshay** [comment]
This would be good to look into. My main question is how does it affect the performance?

**Reply by firxworx** [reply]
I just saw your question now my apologies, because I saw notification about the question I was tagged in below. Its very performant, rust-based, barely there. 

It actually makes for a convenient dev experience too where you can run node locally, have a node container for other deployment scenarios (if required), and run basically the same thing deployed to lambda. I used the lambda container image deployment back on the project I was working on when I suggested this, worked great.

**Comment by mseele** [comment]
@firxworx did you ever tried that variant with astro.js ssr? I'm courious if it's faster than a node lambda with aws adapter!?

**Reply by firxworx** [reply]
It is highly performant, its written in rust. I first started using it when nobody had heard of it. Give it a shot. I haven't benchmarked the different options so I can't answer your specific question.
